### ISSUE TITLE
ALB-096: Migrate training checkpoints from SafeTensors to APR format

### ISSUE THREAD

**Issue by noahgift** [paiml/albor#71]
## Problem

CUDA trainer saves checkpoints as SafeTensors (`.safetensors`), not APR (`.apr`). This violates the sovereign stack architecture where APR is the universal model format for both training and inference.

Current checkpoint output is **4 separate files**: `model.safetensors`, `config.json`, `optimizer_state.json`, `training_state.json`. Non-atomic — files can become inconsistent between save and load.

## Five Whys

1. **Why SafeTensors?** `CudaTransformerTrainer::save()` hard-codes `SaveConfig::new(ModelFormat::SafeTensors)`
2. **Why hard-coded?** No `ModelFormat::Apr` variant exists in `io/format.rs`
3. **Why no APR variant?** APR writer is in aprender (already a dependency) but never wired into `io/save.rs`
4. **Why is this a problem?** Non-atomic checkpoint (4 files). Config mismatch between YAML spec and saved config goes undetected on resume.
5. **Root cause**: APR is designed for this — atomic single-file, `__training__.*` tensor namespace for optimizer state, rich metadata for config/step/lr/loss, `open_filtered()` for inference-only loading, streaming writer for large models, CRC32 integrity.

## Acceptance Criteria

- [ ] `CudaTransformerTrainer` saves checkpoints as `.apr` files using `AprV2StreamingWriter`
- [ ] Single atomic file contains: model weights (F32), optimizer state (`__training__.optimizer.*`), training metadata (step, lr, loss, config)
- [ ] `load_transformer_model()` detects and loads `.apr` checkpoints
- [ ] Checkpoint resume from `.apr` produces correct loss (not random init)
- [ ] `prepare_async_save()` uses APR streaming writer for non-blocking checkpoint saves
- [ ] Contract: `training-checkpoint-apr-v1.yaml`
- [ ] Backward compat: SafeTensors loading still works for existing checkpoints

### entrenar changes:
1. Add `ModelFormat::Apr` to `io/format.rs`
2. Add `save_apr()` to `io/save.rs` — uses `AprV2StreamingWriter`
3. Modify `CudaTransformerTrainer::save()` and `prepare_async_save()` to use APR
4. Modify `load_transformer_model()` to detect `.apr` files
5. Embed full `TransformerConfig` in APR metadata
6. Store optimizer state as `__training__.optimizer.{m,v}.{layer_idx}` tensors
7. Store step/lr/loss/epoch in APR metadata

### Tensor naming convention:
```
# Model weights (same as SafeTensors)
model.embed_tokens.weight
model.layers.0.self_attn.q_proj.weight
...

# Training state (APR __training__ namespace)
__training__.optimizer.embed.m
__training__.optimizer.embed.v
__training__.optimizer.step
__training__.lm_head.m
__training__.lm_head.v
__training__.final_norm.m
__training__.final_norm.v
```

Refs #61
